### PR TITLE
callback basics assignment

### PR DESCRIPTION
**Pull request by blockchainguyy**

### CODE REVIEW

**Review by devanshutechracers — CHANGES_REQUESTED** [review]
Files to run the code are missing.

File: assignment4.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+fs = require('fs');
+function callBackExample(file){
+  fs.readFile(file[i], 'utf8' , function (err,data) {
+    if (err) {
+	  return console.log(err);
+	}
+	console.log(data);
+	console.log('==================');
+	i++;
+	if(i<5) {
```
Comment (rockon1985) [inline assignment4.js]: there should be a space before and after an operator in JS. Eg: `a + b` instead of `a+b`

Comment (rockon1985) [inline assignment4.js]: there should be a space before and after an operator in JS. eg `a + b` instead of `a+b`

File: assignment4.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+let files = ["./f1","./f2","./f3", "./f4", "./f5"], i = 0;
+fs = require('fs');
+function callBackExample(file){
```
Comment (rockon1985) [inline assignment4.js]: rename the method to something meaningful like readFiles or something

File: assignment4.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	console.log(data);
+	console.log('==================');
+	i++;
+	if(i<5) {
+	  callBackExample(file);
+	}
+  });
+
+}
+callBackExample(files);
```
Comment (rockon1985) [inline assignment4.js]: add a new line at the end of each file

File: assignment4.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+let files = ["./f1","./f2","./f3", "./f4", "./f5"], i = 0;
+fs = require('fs');
+function callBackExample(file){
+  fs.readFile(file[i], 'utf8' , function (err,data) {
+    if (err) {
+	  return console.log(err);
```
Comment (rockon1985) [inline assignment4.js]: correct indentation in this file. Required indentation: two spaces

File: assignment4.js (hunk 2 above)
Comment (rockon1985) [reply assignment4.js]: > file parameter name seems as if its a single file, but it's actually a list of files. Rename the parameter accordingly

**Review by rockon1985 — COMMENTED** [review]
Correct indentation and improve nomenclature

**Review by rockon1985 — APPROVED** [review]

File: assignment4.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+let files = ["./f1","./f2","./f3", "./f4", "./f5"], i = 0;
```
Comment (devanshutechracers) [inline assignment4.js]: what is `i`, could this be elaborate name?

File: assignment4.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+fs = require('fs');
```
Comment (devanshutechracers) [inline assignment4.js]: `fs` should be defined. using global namespace is not supported by all NodeJS version.

File: assignment4.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+function fileReader(file){
```
Comment (devanshutechracers) [inline assignment4.js]: There should be a space between starting the function and the curly braces.

File: assignment4.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  fs.readFile(fileList[i], 'utf8' , function (err,data) {
+    if (err) {
+    return console.log(err);
```
Comment (devanshutechracers) [inline assignment4.js]: indentation from line 6-13 is messed up

File: assignment4.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+let files = ["./f1","./f2","./f3", "./f4", "./f5"], i = 0;
+fs = require('fs');
+function fileReader(file){
+  fs.readFile(fileList[i], 'utf8' , function (err,data) {
```
Comment (devanshutechracers) [inline assignment4.js]: there should only be a space after comma, not before while calling a function

File: assignment4.js (hunk 6 above)
Comment (devanshutechracers) [inline assignment4.js]: All the import should be done in the first line.

File: assignment4.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+  }
+  console.log(data);
+  console.log('==================');
+  i++;
+  if(i < 5) {
+    callBackExample(fileList);
+  }
+  });
+}
+fileReader(files);
```
Comment (devanshutechracers) [inline assignment4.js]: line break after finishing a function.

File: assignment4.js (hunk 9 above)
Comment (devanshutechracers) [inline assignment4.js]: What is `fileList`? This is a Runtime Error, did you run the code before submission.

**Review by devanshutechracers — CHANGES_REQUESTED** [review]
Runtime Error. Please run your code before submission.

File: assignment4.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+    console.log(data);
+    console.log('==================');
+    fileNo++;
+    if(fileNo < 5) {
+      fileReader(fileList);
+    }
+  });
+}
+
+fileReader(files);
```
Comment (devanshutechracers) [inline assignment4.js:20]: EOF should contain New Line.

File: assignment4.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
+const fs = require('fs');
+const files = ["./f1","./f2","./f3", "./f4", "./f5"];
+let fileNo = 0;
+function fileReader(fileList) {
+  fs.readFile(fileList[fileNo], 'utf8', function (err,data) {
```
Comment (devanshutechracers) [inline assignment4.js]: space after `,` is required.

File: assignment4.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
+const fs = require('fs');
+const files = ["./f1","./f2","./f3", "./f4", "./f5"];
```
Comment (devanshutechracers) [inline assignment4.js]: Line break should be added after requires

File: assignment4.js [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+let fileNo = 0;
+function fileReader(fileList) {
```
Comment (devanshutechracers) [inline assignment4.js]: Line break before function would increase readability.

**Review by devanshutechracers — CHANGES_REQUESTED** [review]
Add Line Breaks and space to improve readability.

**Review by devanshutechracers — APPROVED** [review]
Approved.

**Review by rockon1985 — APPROVED** [review]

**Comment by rockon1985** [comment]
Finally satisfied with the code! Merging it!

**Pull request merged**